v2/datastream-to-spanner/src/test/java/com/google/cloud/teleport/v2/templates/DatastreamToSpannerSingleDFShardedMigrationIT.java

```diff
@@ -0,0 +1,276 @@
+@TemplateIntegrationTest(DataStreamToSpanner.class)
+@RunWith(JUnit4.class)
+public class DatastreamToSpannerSingleDFShardedMigrationIT extends DataStreamToSpannerITBase {
+  private static final Logger LOG =
+      LoggerFactory.getLogger(DatastreamToSpannerSingleDFShardedMigrationIT.class);
+
+  private static final String TABLE = "Users";
+
+  private static final String SESSION_FILE_RESOURCE =
+      "DatastreamToSpannerSingleDFShardedMigrationIT/mysql-session.json";
+
+  private static final String SHARDING_CONTEXT_RESOURCE =
+      "DatastreamToSpannerSingleDFShardedMigrationIT/sharding-context.json";
+
+  private static final String SPANNER_DDL_RESOURCE =
+      "DatastreamToSpannerSingleDFShardedMigrationIT/spanner-schema.sql";
+
+  private static HashSet<DatastreamToSpannerSingleDFShardedMigrationIT> testInstances =
+      new HashSet<>();
+  private static PipelineLauncher.LaunchInfo jobInfo;
+
+  public static PubsubResourceManager pubsubResourceManager;
+  public static SpannerResourceManager spannerResourceManager;
+
+  /**
+   * Setup resource managers and Launch dataflow job once during the execution of this test class.
+   *
+   * @throws IOException
+   */
+  @Before
+  public void setUp() throws IOException, InterruptedException {
+    // Prevent cleaning up of dataflow job after a test method is executed.
+    skipBaseCleanup = true;
+    synchronized (DataStreamToSpannerShardedMigrationWithMigrationShardIdColumnIT.class) {
+      testInstances.add(this);
+      if (spannerResourceManager == null) {
+        spannerResourceManager = setUpSpannerResourceManager();
+        createSpannerDDL(spannerResourceManager, SPANNER_DDL_RESOURCE);
+      }
+      if (pubsubResourceManager == null) {
+        pubsubResourceManager = setUpPubSubResourceManager();
+      }
+      createAndUploadJarToGcs("shard1");
+      if (jobInfo == null) {
+        jobInfo =
+            launchDataflowJob(
+                getClass().getSimpleName() + "shard1",
+                SESSION_FILE_RESOURCE,
+                null,
+                "shard1",
+                spannerResourceManager,
+                pubsubResourceManager,
+                new HashMap<>() {
+                  {
+                    put("inputFileFormat", "avro");
+                  }
+                },
+                null,
+                SHARDING_CONTEXT_RESOURCE);
+      }
+    }
+  }
+
+  /**
+   * Cleanup dataflow job and all the resources and resource managers.
+   *
+   * @throws IOException
+   */
+  //  @AfterClass
+  //  public static void cleanUp() throws IOException {
+  //    for (DatastreamToSpannerSingleDFShardedMigrationIT instance : testInstances) {
+  //      instance.tearDownBase();
+  //    }
+  //    ResourceManagerUtils.cleanResources(spannerResourceManager, pubsubResourceManager);
+  //  }
+
+  @Test
+  public void multiShardMigration() {
+    // Two dataflow jobs are running corresponding to two physical shards containing two logical
```

Comment:
Stale comment?

Reply:
Can you update the comment describing how this is testing multiple shards migration with single dataflow job? Is it users responsibility to configure the notification for all the shards to the same Pubsub topic?

Reply:
done